src/aks-preview/azext_aks_preview/managed_cluster_decorator.py

```diff
@@ -3253,6 +3265,25 @@ def set_up_agentpool_profile_ssh_access(self, mc: ManagedCluster) -> ManagedClus
                 agent_pool_profile.security_profile.ssh_access = ssh_access
         return mc
 
+    def set_up_bootstrap_profile(self, mc: ManagedCluster) -> ManagedCluster:
+        self._ensure_mc(mc)
+
+        bootstrap_artifact_source = self.context.get_bootstrap_artifact_source()
+        bootstrap_container_registry_resource_id = self.context.get_bootstrap_container_registry_resource_id()
+        if bootstrap_artifact_source is not None:
+            if bootstrap_artifact_source == CONST_ARTIFACT_SOURCE_DIRECT and bootstrap_container_registry_resource_id:
```

Comment:
what about `bootstrap_artifact_source != CONST_ARTIFACT_SOURCE_CACHE`? There might be other artifact source in the future.